### PR TITLE
Adding capability to send to DSS channels [sc-161131]

### PR DESCRIPTION
**Pull request by liamlynch-data**
- added ability to send to channels - using two implementations of AbstractMessageClient
- removed body encoding so UTF-8 always used on SMTP
- dynamic load of channel options by calling list api
- use the presence/absence of the list method in the UI to determine whether to call it to get channels. For DSS < 12.6, only the Direct SMTP option will appear.
- Appended sender emails to channel options when these exist 
- Make sender UI options disappear when there is a sender specified in the channel - needed to add a suffix in the key for the sender field to allow the visibilityCondition to work with this. Made sure this suffix was v. unlikely to be the end of the name of a real channel  
- Either None/null or "smtp" can indicate a direct stmp field - this means legacy configs are supported
- SMTP settings only visible if direct SMTP selected
- Lots of UI shifting 
- change of description
- improvements to error handing
- added templating variable expansion to the subject field
- changed jinja version for py 3.6 compatibility
- removed older py versions and updated the changelog

Re error handling - some notes in the shortcut card on what to expect

### CODE REVIEW

File: custom-recipes/send-mails-from-contacts-dataset/recipe.py [hunk 1]
```diff
@@ -78,45 +105,76 @@ def send_email_for_contact(mail_client, contacts_row, message_template):
...
+has_subject_column = subject_column and not use_subject_value
+has_subject_value = use_subject_value and subject_value
+if not has_subject_column and not has_subject_value:
+    raise AttributeError("No value provided for the subject")
+
+
+has_sender_column = sender_column and not use_sender_value and not channel_has_sender
+has_sender_value = use_sender_value and sender_value and not channel_has_sender
+if not (channel_has_sender or has_sender_column or has_sender_value):
+    raise AttributeError("No value provided for the sender")
```
Comment (alexbourret) [inline custom-recipes/send-mails-from-contacts-dataset/recipe.py:138]: Maybe there should be a similar check for the recipient column

**Review by alexbourret — COMMENTED** [review]
All good, just that one check on the recipient column

File: custom-recipes/send-mails-from-contacts-dataset/recipe.py (hunk 1 above)
Comment (liamlynch-data) [reply custom-recipes/send-mails-from-contacts-dataset/recipe.py:138]: Done [here](https://github.com/dataiku/dss-plugin-sendmail/pull/9/commits/87a5861fe84296151b32d79bb4dc8e00ead45de2)

**Review by alexbourret — APPROVED** [review]
👍

**Comment by liamlynch-data** [comment]
I also tested against a MS 365 channel and it worked fine

**Pull request merged**